amethyst_rendy/src/prefab.rs

```diff
@@ -0,0 +1,112 @@
+use amethyst_assets::{
+    erased_serde::private::serde::{de, de::SeqAccess, ser::SerializeSeq},
+    prefab::{
+        register_component_type,
+        serde_diff::{ApplyContext, DiffContext},
+        SerdeDiff,
+    },
+    DefaultLoader, Handle, Loader,
+};
+use amethyst_core::ecs::{component, Entity, EntityStore, IntoQuery, Resources, World};
+use rendy::mesh::{MeshBuilder, Normal, Position, Tangent, TexCoord};
+use serde::{Deserialize, Serialize};
+use type_uuid::TypeUuid;
+
+use crate::{shape::Shape, types::MeshData, Mesh};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+/// Vertex format to use when generating the mesh.
+pub enum VertexFormat {
+    /// Use a tuple with Positions and Textures
+    VecPosTex,
+    /// Tuple with Positions, Normals and Textures
+    VecPosNormTex,
+    /// Tuple with Positions, Normals Tangents and Textures
+    VecPosNormTangTex,
+}
+
+/// Component provided to create a Mesh from a simple Shape from inside a Prefab
+#[derive(Debug, Clone, Serialize, Deserialize, TypeUuid)]
+#[uuid = "046b3348-424f-4db2-8888-f01206aa1dad"]
+pub struct ShapePrefab {
+    /// Generate a Mesh from a basic type
+    shape: Shape,
+    /// Vertex format to use
+    format: VertexFormat,
+    /// scale to apply to the shape
+    scale: Option<(f32, f32, f32)>,
+}
+
+impl SerdeDiff for ShapePrefab {
+    fn diff<'a, S: SerializeSeq>(
+        &self,
+        ctx: &mut DiffContext<'a, S>,
+        other: &Self,
+    ) -> Result<bool, <S as SerializeSeq>::Error> {
+        unimplemented!()
+    }
+
+    fn apply<'de, A>(
+        &mut self,
+        seq: &mut A,
+        ctx: &mut ApplyContext,
+    ) -> Result<bool, <A as SeqAccess<'de>>::Error>
+    where
+        A: de::SeqAccess<'de>,
+    {
+        unimplemented!()
+    }
+}
+
+impl Default for ShapePrefab {
+    fn default() -> Self {
+        unimplemented!()
```

Comment:
Should this and the other unimplemented be implemented? This looks like it will cause quite a few errors to users thinking that the type can be created with ::default.